### ISSUE TITLE
[FEATURE] Reference already exported contents by id

### ISSUE THREAD

**Issue by martingraf4** [Baboo7/strapi-plugin-import-export-entries#36]
**Is your feature request related to a problem? Please describe.**
Let's say we have two contents `a` and `b`. `a` references `b`:

```json
{
  "id": "<a's id>",
  "name": "a",
  ...,
  "references": ["<b's id>"]
}
```

`b` references `a`:

```json
{
  "id": "<b's id>",
  "name": "b",
  ...,
  "references": ["<a's id>"]
}
```

Currently, when exporting `a` and `b`, depending on the deepness, their content will be duplicated in the exported file that often:

```json
[
  {
    "id": "<a's id>",
    "name": "a",
    ...,
    "references": [
      {
        "id": "<b's id>",
        "name": "b",
        ...,
        "references": [
          {
            "id": "<a's id>",
            "name": "a",
            ...,
            "references": ["..."]
          }
        ]
      }
    ]
  }
]
```

**Describe the solution you'd like**
Instead of duplicating the same content in the exported file, we should check if the export already contains the exported content and if yes, only add the content by id:
```json
[
  {
    "id": "<a's id>",
    "name": "a",
    ...,
    "references": [
      {
        "id": "<b's id>",
        "name": "b",
        ...,
        "references": ["<a's id>"]
      }
    ]
  }
]
```

**Describe alternatives you've considered**
- Selecting an applicable deepness level does not always apply. In our project, we have deeply nested contents next to contents without references in the same collection type.
- `Export relations as id` does not work if the collection type makes use of dynamiczones as the content of the dynamiczones cannot be exported (currently, the content of dynamiczones is collapsed to an id as well):
```json
[
  {
    "id": "<content with dynamiczone>",
    ...,
    "dynamiczone": ["<id of dynamiczone>"]
  }
]
```

**Additional context**
While it is possible to make it work with lots of tweaking, currently, implementing the above feature brings the following advantages:
- Currently, if there are loops in many-to-many relationships, this can have a big impact on the performance. By only adding every content once, this performance problem is resolved.
- The "deepness" theoretically becomes obsolete as it should not be a problem anymore to go as deep until all transitively referenced contents are exported
- Therefore, the plugin becomes easier to use for users as they do not have to think about these configuration parameters anymore and instead it "just works"

**Comment by Baboo7** [comment]
Hey @martingraf4, FYI I just published version `1.15.0` where v2 of export/import has been implemented. It's a beta still but it showed more stable results than the v1 on the test data I used. The export file is flattened with lead to a reduction of 80% in the size of the exported file. Export v2 is accessible in the export modal by selecting the export format `JSON (v2)` / import v2 is detected automatically from the structure of the provided data file. This new version `1.15.0` does NOT introduce breaking changes so v1 is still accessible and behaving as in previous version 😉 

I would love to have your feedback on this new version. Feel free to join the [discord server](https://discord.gg/cGRJh7M7) of the project I just opened 😌
